main.c

```diff
@@ -2,84 +2,172 @@
+void clearScreen()
+{
+  printf("\033[H\033[J");
 }
-void gotoxy(int x, int y) {
-    printf("\033[%d;%dH", (y), (x));
+void gotoxy(int x, int y)
+{
+  printf("\033[%d;%dH", (y), (x));
 }
 
-
 struct winsize w;
 
 int stop = 0;
 
-void clearLine () {
-    printf("%c[2K", 27);
-    printf("\r");
-    fflush(stdout);
+void printSpecialChar (char* str)
+{
+  printf("%c%s", 27, str);
+  fflush(stdout);
 }
 
-void printString (char* str) {
-    printf("%s", str);
-    fflush(stdout);
+void clearLine ()
+{
+  printSpecialChar("[2K\r");
+  fflush(stdout);
 }
 
-void saveCursorPos () {
-    printf("%c7", 27);
-    fflush(stdout);
+void printString (char* str)
+{
+  printf("%s", str);
+  fflush(stdout);
 }
 
-void restoreCursorPos () {
-    printf("%c8", 27);
-    fflush(stdout);
+void saveCursorPos ()
+{
+  printSpecialChar("7");
 }
 
-void *drawProgress(void *null)
+void restoreCursorPos ()
 {
-    int width = w.ws_col;
-    int height = w.ws_row;
-    int n = 0;
-    while (stop == 0) {
-        usleep(20000);
-        saveCursorPos();
-        gotoxy(n, height);
-        n++;
-        if (n > width - 1) {
-            n = 0;
-            clearLine();
-        }
-        printString("#");
-        restoreCursorPos();
-    }
-    return NULL;
-
+  printSpecialChar("8");
 }
 
-int main () {
-    pthread_t drawProgress_thread;
-    int c;
-
-    clearScreen();
-
-    ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);
-    if(pthread_create(&drawProgress_thread, NULL, drawProgress, NULL)) {
-
-        fprintf(stderr, "Error creating thread\n");
-        return 1;
-
+void redrawPrevLines (char matrix[1000][1000], int currentLine)
+{
+  const int width = w.ws_col;
+  saveCursorPos();
+  gotoxy(0, 0);
+  clearLine();
+  for (int i = 1; i <= currentLine; ++i) {
+    gotoxy(0, i);
+    clearLine();
+    for (int j = 1; j <= width; ++j) {
+      gotoxy(j, i);
+      printf("%c", matrix[i][j]);
     }
-    if(pthread_join(drawProgress_thread, NULL)) {
+  }
+  restoreCursorPos();
+}
 
-        fprintf(stderr, "Error joining thread\n");
-        return 2;
+void *drawProgress (void *null)
+{
+  const int width = w.ws_col;
+  // int height = w.ws_row;
+  int n = 0;
+  while (stop == 0) {
+    usleep(20000);
+    saveCursorPos();
+    // Moves carret one line up and to start of line
+    printSpecialChar("[1A\r");
+    n++;
+    if (n > width - 1) {
+      n = 0;
+      clearLine();
+    }
+    for (int i = 0; i < n; ++i) {
+      printString("#");
+    }
+    restoreCursorPos();
+  }
+  return NULL;
+}
 
+void *proceedInput (void *null)
+{
+  int c;
+  // Matrix shpuld be limited to width and height
+  // But unfortunantely for now it will be 1000 * 1000
+  // int width = w.ws_col;
+  // int height = w.ws_row;
+  char charMatrix[1000][1000];
+  int currentLine = 1;
+  int n = 1;
+  const char quitSeq[4] = "quit";
+
+  int quitSeqIter = 0;
+  while((c = getchar()) != EOF) {
+    printf("%c", c);
+    if (c == quitSeq[quitSeqIter] || c == 27 || c == 24) {
```

Comment:
It's unclear what is 27, 24, please use ASCII symbols.

Reply:
What ascii symbol for escape and cancel?

Reply:
They are unprintable, please define them in global scope and compare against them.
```c == ESC```

Reply:
@ikakok Done Moved them to separate file